simulator.py

```diff
@@ -0,0 +1,62 @@
+# Class - Simulator
+# Runs multiple simulations of the game and saves the results
+
+import game
+import multiprocessing as mp
+import csv
+
+class Simulator:
+    """Class that manages simulations of a BlackJack game
+    
+    Parameters:
+
+    sim_count (int): Indicated the number of simulations to compute
+
+    """
+
+    def __init__(self, sim_count=100, bet_amount=1):
+        self.sim_count = sim_count
+        self.bet_amount = bet_amount
+
+    def main(self):
+        print("Starting Blackjack Simulator")
+        results = []
+        
+        #TODO: Implement concurrency
+        for x in range(self.sim_count):
+            results.append(self.run_game())
+        
+        self.write_results(results)
+        
+        print(f'Result for this simulation: {sum(results)/self.sim_count}')
+        print(f'Adjusted overall result: {self.get_overall_average(results)}')
+
+        # for x in range(10):
```

Comment:
Still reading about how to best do this!

Reply:
For this, I think we can use the concurrent.futures.ProcessPoolExecutor.

I believe it's structured like this
```
with concurrent.futures.ProcessPoolExecutor() as executor:
   p_list = [] # initial empty process list
   for i in range(<number_of_instances>):
      p_list.append(executor.submit(<run_x_games>, 1)) # add process to list

   for f in concurrent.futures.as_completed(p_list):
      print(p_list.result())     # get return result from each completed function call   
```

I think we'll need to pass in a function that would run a certain number of games (ex. lets say it runs 100 games and returns the combined results of those games). Then when we pass that function as the function to be executed by the process. We'll then have multiple runs of 100 games that should be operating concurrently and hopefully that helps cut down on our runtime.

Reply:
I got the structure of the code from this video https://youtu.be/fKl2JW_qrso?t=1061 (should skip to the timestamp).
He does a good job at explaining what does what which I found very helpful.